File: test_function.jl
using Cubature

# Define your function G(r, a, alpha, beta)
function G(x, alpha, beta)
    return alpha * x[1] + beta * x[2]
end

inf = 10000000000000000000000000000000000000000

# Define a function to compute the double integral
function double_integral(alpha_values, beta_values)
    min_integral = Inf
    min_alpha = nothing
    min_beta = nothing
    for alpha in alpha_values
        for beta in beta_values
            # Define the function to integrate
            f(x) = G(x, alpha, beta)
            # Perform the double integral
            integral, error = hcubature(f, [0.0, 0.0], [inf, 2π])
            # Update minimum if a smaller integral is found
            if integral < min_integral
                min_integral = integral
                min_alpha = alpha
                min_beta = beta
            end
        end
    end

    return min_alpha, min_beta, min_integral
end

# Define arrays of alpha and beta values
alpha_values = range(0.1, stop=2.0, length=10)
beta_values = range(0.1, stop=2.0, length=10)

# Compute the minimum integral and corresponding alpha and beta
min_alpha, min_beta, min_integral = double_integral(alpha_values, beta_values)

println("Minimum Integral:", min_integral)
println("Corresponding Alpha:", min_alpha)
println("Corresponding Beta:", min_beta)

File: test_function.py
import numpy as np
from scipy import integrate
from numba import jit, float64
from numba.core.errors import NumbaWarning
import warnings

warnings.simplefilter('ignore', category=NumbaWarning)

@jit(nopython=True)
def G(r: float64, a: float64, alpha: float64, beta: float64):
  # Define your function G(r, a, alpha, beta) here
  # For example: return alpha * r + beta * a
  # Modify this according to your actual function.
  return alpha * r + beta * a

@jit
def double_integral(alpha_values, beta_values):
  min_integral = np.inf
  min_alpha = None
  min_beta = None
  for alpha in alpha_values:
    for beta in beta_values:
      # Define the integrand as a function of r and a
      # integrand = lambda r, a: G(r, a, alpha, beta)

      # Perform the double integral
      integral, _ = integrate.nquad(G, [[0, np.inf], [0, 2*np.pi]], args=(alpha, beta))

      # Update minimum if a smaller integral is found
      if integral < min_integral:
        min_integral = integral
        min_alpha = alpha
        min_beta = beta
  return min_alpha, min_beta, min_integral


# Define arrays of alpha and beta values
alpha_values = np.linspace(0.1, 2.0, 10)
beta_values = np.linspace(0.1, 2.0, 10)

# Compute the minimum integral and corresponding alpha and beta
min_alpha, min_beta, min_integral = double_integral(alpha_values, beta_values)

print("Minimum Integral:", min_integral)
print("Corresponding Alpha:", min_alpha)
print("Corresponding Beta:", min_beta)